File: 2015/day01/python/2015_day01.py
def elevator():
    
    with open("../input.txt", "r") as file:
        data = file.read()

    floor = 0

    for move in data:
        if move == "(":
            floor += 1

        elif move == ")":
            floor -= 1

    print(f"Santa is on floor: {floor}")

if __name__ == "__main__":
    elevator()

File: 2015/day01/c/2015_day01.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    FILE *file = fopen("../input.txt", "r");
    if (file == NULL) {
        printf("Error opening file");
        return 1;
    }

    int floor = 0;
    char move;

    while ((move = fgetc(file)) != EOF) {
        switch (move) {
            case '(':
                floor++;
                break;

            case ')':
                floor--;
                break;

            default:
                break;
        }
    }

    printf("Santa is on floor: %d\n", floor);

    fclose(file);
    
    return 0;
}